src/memory.rs

```diff
@@ -114,7 +116,30 @@ impl Memory {
             unit: scaled_unit,
         }
     }
+
+    /// Scale up or down to the desired `binary_multiple`. Returns a new `Memory` and does
+    /// not change itself.
+    pub fn scale_to(&self, binary_multiple: BinaryMultiple) -> Self {
+        let self_discriminant = self.unit as i32;
+        let to_discriminant = binary_multiple as i32;
+
+        let diff: i32 = (self_discriminant - to_discriminant) as i32;
+
+        Memory {
+            value: self.value * 1024f32.powi(diff),
+            unit: binary_multiple,
+        }
+    }
+
+    pub fn value(&self) -> f32 {
```

Comment:
As discussed, these getter don't look rusty.

Reply:
Yes ill remove them and add a to_java_heap() method directly on the Memory struct?
I think i still want to return `Result<(u32, BinaryMultiple), Error>` seperatly in case any product requires it?